modules.d/99squash/module-setup.sh

```diff
@@ -25,7 +25,16 @@ installpost() {
     # Move everything under $initdir except $squash_dir
     # itself into squash image
     for i in "$initdir"/*; do
-        [[ $squash_dir == "$i"/* ]] || mv "$i" "$squash_dir"/
+        [[ $squash_dir == "$i"/* ]] || [[ "$initdir/etc" == "$i" ]] || mv "$i" "$squash_dir"/
+    done
+    # keep ld cache under $initdir
+    mkdir -p "$squash_dir"/etc
+    for i in "$initdir"/etc/*; do
+        if [[ $i =~ "$initdir"/etc/ld.so* ]]; then
+            cp -r "$i" "$squash_dir"/etc
+        else
+            mv "$i" "$squash_dir"/etc
+        fi
```

Comment:
Hi Pingfan,

I think we can just add one line of code here: `inst /etc/ld.so.cache`, `/etc/ld.so.cache` is the only file affects early squash loader. No need to move&copy things again and again.